cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/strategy/AbstractReplicaMovementStrategy.java

```diff
@@ -23,17 +23,7 @@
   @Override
   public ReplicaMovementStrategy chain(ReplicaMovementStrategy strategy) {
     AbstractReplicaMovementStrategy current = this;
-    return new ReplicaMovementStrategy() {
-      @Override
-      public Map<Integer, SortedSet<ExecutionTask>> applyStrategy(Set<ExecutionTask> replicaMovementTasks, Cluster cluster) {
-        return current.applyStrategy(replicaMovementTasks, cluster);
-      }
-
-      @Override
-      public ReplicaMovementStrategy chain(ReplicaMovementStrategy strategy) {
-        return current.chain(strategy);
-      }
-
+    return new AbstractReplicaMovementStrategy() {
```

Comment:
This is all about OOP. The way the `ReplicaMovementStrategy` was used is a composition. It didn't use the overridden `taskComparator` method at all, because `current.applyStrategy` would be called on a separate object, not on an instance of `ReplicaMovementStrategy`.